pkg/mcs/tso/server/server.go

```diff
@@ -510,8 +561,11 @@ func (s *Server) startServer() (err error) {
 		return err
 	}
 
+	serverReadyChan := make(chan struct{})
```

Comment:
why add it?

Reply:
To ensure tso.Server.StartServer() successfully exits right before cmux.CMux.Serve() is called. Without this, the TestServerRegisterTestSuite.TestServerPrimaryChange will fail, because it starts tso server then call tso.Server.Close() immediately. Generally, it's better to shorten the distance between tso.Server.StartServer() successfully returned and cmux.CMux.Serve() being called.